### PR TITLE
Review:

### PR DESCRIPTION
**Pull request by ViliusKraujutis**
* Buttons do what their title says
* Extract button click listeners to single anonymous parameterized class
* Connect to controller only when application is visible. Disconnect every time application is hidden.
* Don't turn off lights when connection is established

### CODE REVIEW

File: app/src/main/java/lt/andro/metasocket/MainActivity.java [hunk 1]
```diff
@@ -40,52 +42,42 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        buttonOff.setOnClickListener(turnOff);
+    }
 
-        ///< Bind the service when the activity is created
-        getApplicationContext().bindService(new Intent(this, MetaWearBleService.class),
-                this, Context.BIND_AUTO_CREATE);
+    @Override
+    protected void onPause() {
+        super.onPause();
+        ///< Unbind the service when the activity is hidden
```
Comment (2m) [inline app/src/main/java/lt/andro/metasocket/MainActivity.java:55]: Never seen comments like this before. :)

**Review by 2m — APPROVED** [review]
LGTM

File: app/src/main/java/lt/andro/metasocket/MainActivity.java [hunk 2]
```diff
@@ -40,52 +42,42 @@ protected void onCreate(Bundle savedInstanceState) {
...
-                } catch (UnsupportedModuleException e) {
-                    e.printStackTrace();
-                }
-            }
-        });
+        buttonOn.setOnClickListener(turnOn);
+        buttonOff.setOnClickListener(turnOff);
+    }
 
-        ///< Bind the service when the activity is created
```
Comment (ViliusKraujutis) [inline app/src/main/java/lt/andro/metasocket/MainActivity.java:67]: @2m, I've just updated created/destroyed to resumed/paused, so connection would be held shorter.

File: app/src/main/java/lt/andro/metasocket/MainActivity.java (hunk 1 above)
Comment (ViliusKraujutis) [reply app/src/main/java/lt/andro/metasocket/MainActivity.java:55]: See this [comment](https://github.com/andro-lt/metasocket/pull/1#pullrequestreview-9407520). That's a copy of previous comment.

**Pull request merged**